frontend/src/layout/header/Search.tsx

```diff
@@ -61,6 +61,17 @@ export const SearchField: React.FC<SearchFieldProps> = ({ variant }) => {
         ? new URLSearchParams(document.location.search).get("q") ?? undefined
         : undefined;
 
+    const search = (expression: string) => {
+        if (lastTimeout.current !== null) {
+            clearTimeout(lastTimeout.current);
+        }
+        lastTimeout.current = setTimeout(() => {
```

Comment:
Should this timeout logic really be part of `search`? Wouldn't this result in a delay when I press enter to search? 🤔

Reply:
I would still advocate for not incurring an unnecessary delay. Your second point is a good one, though, so an enter press should probably clear the pending timeout. :thinking:

Reply:
I have no strong opinion about that, so I removed the delay.